### PR TITLE
add rule Limit db query results

### PR DESCRIPTION
**Pull request by elisecodedestrucs**
add a rule that forbids SQL requests that can have an infinite number of returns

### CODE REVIEW

**Review by dedece35 — CHANGES_REQUESTED** [review]
Hi @elisecodedestrucs 
Thank you for your PR ...
I let our JS expert review your PR (@utarwyn), but for me can you add an update of `CHANGELOG.md` file with your modification
One more thing, do you upgrade our real test repository [ecoCode-javascript-test-project](https://github.com/green-code-initiative/ecoCode-javascript-test-project) with a test use case (as your unit test) ?

File: eslint-plugin/lib/rules/limit-db-query-results.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+"use strict";
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: "suggestion",
+    docs: {
+      description: "Warn about not limiting the number of returns for a query",
```
Comment (utarwyn) [inline eslint-plugin/lib/rules/limit-db-query-results.js:24]:
```suggestion
      description: "Should limit the number of returns for a SQL query",
```

File: eslint-plugin/lib/rules/limit-db-query-results.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if (typeof node.value == "string") {
+          const query = node.value.toUpperCase();
+          if (
+            query.includes("SELECT") &&
+            query.includes("FROM") &&
+            !(
+              query.includes("LIMIT") ||
+              query.includes("TOP") ||
+              query.includes("ROW_NUMBER") ||
+              query.includes("FETCH FIRST")
```
Comment (utarwyn) [inline eslint-plugin/lib/rules/limit-db-query-results.js:46]: Maybe store the list of keywords in a constant above?
The goal is to simplify the addition of keywords that limit number of returns in the future :+1:

File: CHANGELOG.md [hunk 3]
```diff
@@ -38,6 +38,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -   Setup mocha and nyc for tests and coverage
 -   Setup basic coding style tools
 -   Write complete contributing guide
+-   Add rule `@ecocode/limit-db-query-results`
```
Comment (utarwyn) [inline CHANGELOG.md:41]: Can you put this line in the `unreleased` > `Added` section above? Instead of here

**Review by utarwyn — COMMENTED** [review]
Thank you for this first contribution! Great work 💯 
Just a few comments/proposals to make everything 100% clean..

Maybe you can also create an issue with the list the rules you've added to the ESLint plugin that will need to be added to the SonarQube plugin too

**Comment by dedece35** [comment]
HI @elisecodedestrucs, please correct the "branch out-of-date" warning above :p

**Comment by elisecodedestrucs** [comment]
Hi, Thank you for your returns! 
I have made the requested changes don't hesitate to tell me if there's more

**Comment by utarwyn** [comment]
@elisecodedestrucs can you also re-run npm script `update:eslint-docs` to update documentation files with the new description I proposed? Apart from that, everything looks good to me!

**Review by utarwyn — APPROVED** [review]

**Review by dedece35 — APPROVED** [review]

**Review by dedece35 — CHANGES_REQUESTED** [review]
Hi @elisecodedestrucs 
could you please upgrade https://github.com/green-code-initiative/ecoCode-javascript-test-project repository with a test case ?

**Pull request merged**